runsc/cgroup/cgroup.go

```diff
@@ -214,6 +214,39 @@ func LoadPaths(pid string) (map[string]string, error) {
 	if err := scanner.Err(); err != nil {
 		return nil, err
 	}
+
+	// For nested containers, in /proc/self/cgroup we see paths from host,
+	// which don't exist in container, so recover the container paths here by
+	// double-checking with /proc/pid/mountinfo
+	mountinfo, err := os.Open(filepath.Join("/proc", pid, "mountinfo"))
+	if err != nil {
+		return nil, err
+	}
+	defer mountinfo.Close()
+
+	mfScanner := bufio.NewScanner(mountinfo)
+	for mfScanner.Scan() {
+		txt := mfScanner.Text()
+		fields := strings.Fields(txt)
+		if len(fields) < 9 || fields[len(fields)-3] != "cgroup" {
+			continue
+		}
+		for _, opt := range strings.Split(fields[len(fields)-1], ",") {
+			// we have a subsystem here
+			if cgroupPath, ok := paths[opt]; ok {
+				root := fields[3]
+				relCgroupPath, err := filepath.Rel(root, cgroupPath)
```

Comment:
Use cgroupPath if Rel() fails?

Reply:
I think Rel failed if we have mismatched in mountinfo and cgroup, which means the setup is wrong. I don't think there's any point in creating a wrong hierrachy. IIRC in runc this is also a hard-fail.